xwiki-commons-tools/xwiki-commons-tool-xar/xwiki-commons-tool-xar-plugin/src/main/java/org/xwiki/tool/xar/XARMojo.java

```diff
@@ -222,7 +222,7 @@ private void performTransformations() throws Exception
                                 document = xmlDocument;
                             } else {
                                 List<Element> siblings = parent.elements();
-                                siblings.set(parent.indexOf(node), xmlElement);
+                                siblings.set(siblings.indexOf(node), xmlElement);
```

Comment:
I would add a comment to explain that dom4j ``Node`` doesn't have a ``replaceChild`` method (like ``org.w3c.dom.Node``) so we're forced to replace the element by index. Moreover, dom4j ``Element`` doesn't have a method to get all child nodes (including text or comment nodes) so the index we use must be from the list of child elements.

Reply:
Thank you Marius. Done in below commit (6bd21042c7adcde067d1786ccedcb0cd91de6002).